lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -37,6 +53,13 @@ def matches?(given_proc, negative_expectation=false, &block)
           @eval_block = false
           @eval_block_passed = false
 
+          if @warn_about_bare_error && !negative_expectation && @block.nil?
+            RSpec.warning("Using the `raise_error` matcher without providing a specific " \
+                          "error or message risks false positives, consider providing " \
+                          "a specific error instead. This message can be suppressed using " \
+                          "`RSpec::Matchers::BuiltIn::RaiseError.allow_bare_raise_error = true`")
+          end
```

Comment:
It's clear to us why there's a risk of false positives, but I'm not sure it'll be clear to the user.  Might it be worth fleshing that out a bit?  Something like:

> ...since `raise_error` will match when Ruby raises a `NoMethodError`, `NameError` or `ArgumentError`, potentially allowing the expectation to pass without even executing the method you are intending to call.